### PR TITLE
Hia 687 urls

### PR DESCRIPTION
**Pull request by AlexHerbertMoJ**
#### Context

- Adding URL generation for an event notification

#### Changes proposed in this PR

-

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt [hunk 1]
```diff
@@ -35,4 +35,19 @@ class EventNotification(
   @Temporal(value = TemporalType.TIMESTAMP)
   @Column(name = "LAST_MODIFIED_DATETIME", nullable = false)
   val lastModifiedDateTime: LocalDateTime,
-)
+) {
+  override fun equals(other: Any?): Boolean {
+    if (other == null) return false
```
Comment (CslBing) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt:40]: What happens if `this` is null?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt [hunk 2]
```diff
@@ -35,4 +35,19 @@ class EventNotification(
...
+    if (this === other) return true
+    if (other !is EventNotification) return false
+    if (eventId != other.eventId ||
+      hmppsId != other.hmppsId ||
+      eventType != other.eventType ||
+      url != other.url ||
+      lastModifiedDateTime != other.lastModifiedDateTime
+    ) {
+      return false
+    }
```
Comment (CslBing) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt:50]: Isn't line 43-49 doing the same check as line 41?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/services/RegistrationEventsService.kt [hunk 3]
```diff
@@ -26,19 +26,21 @@ class RegistrationEventsService(
   fun execute(hmppsDomainEvent: HmppsDomainEvent) {
     val registrationEventMessage: RegistrationAddedEventMessage = objectMapper.readValue(hmppsDomainEvent.message)
 
+    // TODO Autowire this in from application.yml
+    val baseUrl = "https://dev.integration-api.hmpps.service.justice.gov.uk"
```
Comment (CslBing) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/services/RegistrationEventsService.kt:30]: TODO?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt (hunk 1 above)
Comment (AlexHerbertMoJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt:40]: The equals method isn't a static method, so surely `this `can't be null for equals to be called?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt (hunk 2 above)
Comment (AlexHerbertMoJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/repository/model/data/EventNotification.kt:50]: no, 41 is testing whether they're actually the same object, wheras 43 - 49 are testing that the two objects have identical parameters, which we're testing for to determine equality

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/services/RegistrationEventsService.kt (hunk 3 above)
Comment (AlexHerbertMoJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/services/RegistrationEventsService.kt]: oops, missed that. Fixed in latest commit

**Pull request merged**